### DISCUSSION TITLE
How do I avoid getting my IP banned.

### DISCUSSION THREAD

**Discussion by omer921** [davidteather/TikTok-Api#438] [Q&A] [answered]
I am working on a project that will require me to scrape a lot of data from TikTok. I was wondering how far some have you guys had pushed the number of requests per minute, hour, day, etc. I'd also like to know what proxy's you guys are using and how much you are paying for them.

Thanks!

**Comment by davidteather** [comment, marked as answer]
Hi! I think they only started to ban residential IPs recently. I would recommend just using a proxy so you don't get detected. I've used [luminati.io](https://luminati.io/) they have pretty good rates and the data-center IPs are enough to get the job done. (and if you burn them you can just refresh the IP pool).
